### PR TITLE
Create books list

### PR DESCRIPTION
**Pull request by Ksenaaa**

### CODE REVIEW

File: src/component/bookList/BookItem.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        setFavoriteBook(favoriteBook === true ? false : true)
+    }, [])
+    
+    return(
+        <div className="wrapperItem">
+            <div className="bookImg">
+                <img src={`http://localhost:3002/${book.img}`} />
+            </div>
+            <div className="bookText">
+                <div className="bookName" onClick={() => navigate(`'/book' + '/' + ${book.idBook}`)}>
```
Comment (valentinPortside) [inline src/component/bookList/BookItem.tsx]: add `/book` to router constants, and use it 
`/book/${book.idBook}`

File: src/component/bookList/BookList.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import './style.css'
+
+type Props = {
+    books: IBook[]
+}
+
+export const BookList: FC<Props> = ({books}) => {
+
+    return(
+        <div className="wrapperList">
```
Comment (valentinPortside) [inline src/component/bookList/BookList.tsx]: wrapperBookList - entity must be identified

File: src/component/bookList/style.css [hunk 3]
```diff
@@ -0,0 +1,99 @@
+.wrapperList {
```
Comment (valentinPortside) [inline src/component/bookList/style.css]: the same as above

File: src/component/bookList/style.css [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    border: 2px solid rgb(209, 205, 205);
+    margin: 30px auto;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: space-around;
+    padding: 10px 30px;
+}
+
+.wrapperItem {
```
Comment (valentinPortside) [inline src/component/bookList/style.css]: the same as above

File: src/component/select/SelectOption.tsx [hunk 5]
```diff
@@ -1,6 +1,10 @@
 import React, { FC, ReactElement, SelectHTMLAttributes } from 'react';
 
-export const SelectOption: FC<SelectHTMLAttributes<ReactElement>> = ({value}) => {
+type Props = {
+  value?: string | number,
+}
+
+export const SelectOption: FC<Props> = ({value}) => {
 
   return (
```
Comment (valentinPortside) [inline src/component/select/SelectOption.tsx]: remove `return` and use  SelectOption: FC<Props> = ({value}) => <option>...</option>

File: src/hooks/book.hook.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+   img: null,
+   sellerId: '',
+}
+
+export const useBook = () => {
+   const bookData = JSON.parse(localStorage.getItem(storageName) as string) || initialData
+
+   const [book, setBook] = useState<IBook>({...bookData})
+
+   const bookLocalStor = useCallback(({_id, bookName, author, category, page, year, language, price, img, sellerId}: IBookResponce) => {
```
Comment (valentinPortside) [inline src/hooks/book.hook.ts]: replace all props to  `book`

File: src/hooks/book.hook.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      const bookInfo = {
```
Comment (valentinPortside) [inline src/hooks/book.hook.ts]: use  const bookInfo = {idBook:book._id, ...book}

File: src/pages/auth/components/ModalRegistration.tsx [hunk 8]
```diff
@@ -19,9 +19,10 @@ type RegisterUser = {
     role: string,
 }
 
-export const ModalRegistration: FC<ModalWindow> = ({onClose}) => {
-    const [form, setForm] = useState<RegisterUser>({email: '', name: '', password: '', checkPassword: '', role: 'buyer'})
-    const {loading, request, errorsValid} = useHttp()
+export const ModalRegistration: FC<Props> = ({ onClose }) => {
+    const [form, setForm] = useState<RegisterUser>({ email: '', name: '', password: '', checkPassword: '', role: 'buyer' })
```
Comment (valentinPortside) [inline src/pages/auth/components/ModalRegistration.tsx:23]: do we really need initial State?

File: src/pages/booksSeller/BooksSeller.tsx [hunk 9]
```diff
@@ -1,7 +1,34 @@
...
+import { AuthContext } from '../../context/AuthContext'
+import { useHttp } from '../../hooks/http.hook'
+import { IBook } from '../../interface/IBook'
+import { normalizeBooks } from '../../utils/normalizeBooks'
 
 export const BooksSeller = () => {
+    const [books, setBooks] = useState<IBook[]>()
+
+    const {userAuth} = useContext(AuthContext)
+
```
Comment (valentinPortside) [inline src/pages/booksSeller/BooksSeller.tsx:18]: remove space between hooks

File: src/pages/booksSeller/BooksSeller.tsx [hunk 10]
```diff
@@ -1,7 +1,34 @@
...
+    }, [])
+
+    useEffect(() => {
+        showBooks()
+    }, [])
+
     return (
-        <div>My Books</div>
+        <>
+        <h1>My books</h1>
```
Comment (valentinPortside) [inline src/pages/booksSeller/BooksSeller.tsx]: spaces!

File: src/pages/createBook/CreateBook.tsx [hunk 11]
```diff
@@ -12,32 +12,24 @@ import { AuthContext } from '../../context/AuthContext';
...
+
     const {userAuth} = useContext(AuthContext)
+
     const {loading, request, errorsValid, clearError} = useHttp()
+    
     const closeCreatePage = useNavigate()
 
     const changeHandler = useCallback(({ target }: ChangeEvent<HTMLInputElement>): void => {
         const value = (target.type === 'file') ? target?.files?.[0] : 
             (target.name === 'price') ? target.value.replace(/[^0-9.]/g, '').replace(/^(\d[^.]*\.)|\./g, '$1') :
```
Comment (valentinPortside) [inline src/pages/createBook/CreateBook.tsx]: (target.name === 'price') = target.name === 'price'

File: src/pages/main/Main.tsx [hunk 12]
```diff
@@ -1,7 +1,30 @@
...
+    }, [])
+
+    useEffect(() => {
+        showBooks()
+    }, [])
+
     return (
-        <div>Main</div>
+        <>
+        {loading && <LoadingCircular />}
```
Comment (valentinPortside) [inline src/pages/main/Main.tsx]: spaces

File: src/utils/normalizeBooks.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import { IBookResponce } from "../interface/IBook";
+
+export const normalizeBooks = (books: IBookResponce[]) => books.map((book) => ({
+    idBook: book._id,
+    ...book
+}))
```
Comment (valentinPortside) [inline src/utils/normalizeBooks.ts]: add space to the end of file

File: src/component/bookList/BookItem.tsx (hunk 1 above)
Comment (Ksenaaa) [reply src/component/bookList/BookItem.tsx]: fixed

File: src/component/bookList/BookList.tsx (hunk 2 above)
Comment (Ksenaaa) [reply src/component/bookList/BookList.tsx]: fixed

File: src/component/bookList/style.css (hunk 3 above)
Comment (Ksenaaa) [reply src/component/bookList/style.css]: fixed

File: src/component/bookList/style.css (hunk 4 above)
Comment (Ksenaaa) [reply src/component/bookList/style.css]: fixed

File: src/pages/auth/components/ModalRegistration.tsx (hunk 8 above)
Comment (Ksenaaa) [reply src/pages/auth/components/ModalRegistration.tsx:23]: really!

File: src/pages/booksSeller/BooksSeller.tsx (hunk 10 above)
Comment (Ksenaaa) [reply src/pages/booksSeller/BooksSeller.tsx]: fixed

File: src/pages/booksSeller/BooksSeller.tsx (hunk 9 above)
Comment (Ksenaaa) [reply src/pages/booksSeller/BooksSeller.tsx:18]: discused

File: src/pages/createBook/CreateBook.tsx (hunk 11 above)
Comment (Ksenaaa) [reply src/pages/createBook/CreateBook.tsx]: fixed

File: src/pages/main/Main.tsx (hunk 12 above)
Comment (Ksenaaa) [reply src/pages/main/Main.tsx]: fixed

File: src/utils/normalizeBooks.ts (hunk 13 above)
Comment (Ksenaaa) [reply src/utils/normalizeBooks.ts]: fixed

File: src/component/select/SelectOption.tsx (hunk 5 above)
Comment (Ksenaaa) [reply src/component/select/SelectOption.tsx]: fixed

File: src/hooks/book.hook.ts (hunk 6 above)
Comment (Ksenaaa) [reply src/hooks/book.hook.ts]: fixed

File: src/hooks/book.hook.ts (hunk 7 above)
Comment (Ksenaaa) [reply src/hooks/book.hook.ts]: fixed

File: .env [hunk 14]
```diff
@@ -0,0 +1 @@
+REACT_APP_API_URL='http://localhost:3002/'
```
Comment (valentinPortside) [inline .env]: add space to the end of file

File: src/component/bookList/BookList.tsx [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+import './style.css'
+
+type Props = {
+    books: IBook[],
+    onScroll?: (e: UIEvent<HTMLDivElement>) => void,
+}
+
+export const BookList: FC<Props> = ({books, onScroll}) => {
+
+    return(
```
Comment (valentinPortside) [inline src/component/bookList/BookList.tsx]: remove return

File: src/component/radioButton/RadioButtonRegistration.tsx [hunk 16]
```diff
@@ -4,12 +4,12 @@ import RadioGroup from '@material-ui/core/RadioGroup';
...
-type RadioButtonType = {
+type Props = {
     onChange(e: ChangeEvent<HTMLInputElement>): void,
     name: string, 
 }
 
-export const RadioButtonRegistration: FC<RadioButtonType> = ({onChange, name}) => {
+export const RadioButtonRegistration: FC<Props> = ({onChange, name}) => {
 
     return (
```
Comment (valentinPortside) [inline src/component/radioButton/RadioButtonRegistration.tsx]: if don't have any logic we write `Components` without return! as `SelectOption`

File: src/constants/pageLimit.ts [hunk 17]
```diff
@@ -0,0 +1 @@
+export const DEFAULT_PAGE_LIMIT = 12
```
Comment (valentinPortside) [inline src/constants/pageLimit.ts]: space

File: src/hooks/book.hook.ts [hunk 18]
```diff
@@ -0,0 +1,39 @@
+import React, { useCallback, useState } from 'react'
+
+import { IBook, IBookResponce } from '../interface/IBook'
+import { normalizeBook } from '../utils/normalizeBooks'
+
+const storageName = 'bookData'
+
+const initialData = {
```
Comment (valentinPortside) [inline src/hooks/book.hook.ts:8]: move it to const file

File: src/interface/IBook.ts [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    price: string,
+    img: File | null,
+    sellerId: string,
+}
+
+export interface IBookResponce extends Omit<IBook, 'idBook'> {
+    _id: string,
+}
+
+export interface IFormBook extends Omit<IBook, 'idBook'> {
```
Comment (valentinPortside) [inline src/interface/IBook.ts]: what is this? why do we need empty interface?

File: src/interface/IRoute.ts [hunk 20]
```diff
@@ -4,26 +4,28 @@ import { Main } from "../pages/main/Main";
...
+    BOOK_PAGE = '/book',
 }
 
 export const publicRoutes: IRoute[] = [
-    {path: RouteNames.LOGIN, exact: true, element: Auth}
+    {path: RouteNames.LOGIN, element: Auth}
 ]
 
 export const privateRoutes: IRoute[] = [
-    {path: RouteNames.MAIN, exact: true, element: Main},
```
Comment (valentinPortside) [inline src/interface/IRoute.ts:26]: return exact?!

File: src/pages/auth/components/ModalLogin.tsx [hunk 21]
```diff
@@ -17,17 +17,19 @@ type LoginUser = {
...
+    const [form, setForm] = useState<LoginUser>({ email: '', password: '' })
 
-    const changeHandler = useCallback((event: ChangeEvent<HTMLInputElement>): void => {
-        setForm({...form, [event.target.name]: event.target.value})
+    const { login } = useContext(AuthContext)
+    
+    const { loading, request, errorsValid } = useHttp()
+
+    const handlerChange = useCallback((event: ChangeEvent<HTMLInputElement>): void => {
+        setForm({ ...form, [event.target.name]: event.target.value })
```
Comment (valentinPortside) [inline src/pages/auth/components/ModalLogin.tsx]: setForm(prevForm => { ...prevForm, [event.target.name]: event.target.value })

File: src/pages/auth/components/ModalLogin.tsx [hunk 22]
```diff
@@ -17,17 +17,19 @@ type LoginUser = {
...
     }, [form])
```
Comment (valentinPortside) [inline src/pages/auth/components/ModalLogin.tsx]: remove `form` from dependency

File: src/pages/createBook/components/InputUploadImg.tsx [hunk 23]
```diff
@@ -35,26 +35,26 @@ export const InputUploadImg: FC<InputFileType> = ({onChange, errors, type, name,
 
     return (
         <>
```
Comment (valentinPortside) [inline src/pages/createBook/components/InputUploadImg.tsx:37]: we don't need `<>` anymore, couse we have `div` wrapper

File: src/utils/parceStringToNumberAndPoint.ts [hunk 24]
```diff
@@ -0,0 +1,2 @@
+export const parceStringToNumberAndPoint = (value: string) => 
+    value.replace(/[^0-9.]/g, '').replace(/^(\d[^.]*\.)|\./g, '$1')
```
Comment (valentinPortside) [inline src/utils/parceStringToNumberAndPoint.ts]: space

File: .env (hunk 14 above)
Comment (Ksenaaa) [reply .env]: fixed

File: src/constants/pageLimit.ts (hunk 17 above)
Comment (Ksenaaa) [reply src/constants/pageLimit.ts]: fixed

File: src/interface/IBook.ts (hunk 19 above)
Comment (Ksenaaa) [reply src/interface/IBook.ts]: removed interface

File: src/interface/IRoute.ts (hunk 20 above)
Comment (Ksenaaa) [reply src/interface/IRoute.ts:26]: fixed

File: src/pages/auth/components/ModalLogin.tsx (hunk 21 above)
Comment (Ksenaaa) [reply src/pages/auth/components/ModalLogin.tsx]: fixed

File: src/component/bookList/BookList.tsx (hunk 15 above)
Comment (Ksenaaa) [reply src/component/bookList/BookList.tsx]: fixed

File: src/pages/auth/components/ModalLogin.tsx (hunk 22 above)
Comment (Ksenaaa) [reply src/pages/auth/components/ModalLogin.tsx]: fixed

File: src/pages/createBook/components/InputUploadImg.tsx (hunk 23 above)
Comment (Ksenaaa) [reply src/pages/createBook/components/InputUploadImg.tsx:37]: No, we need them. They wrap two div

File: src/utils/parceStringToNumberAndPoint.ts (hunk 24 above)
Comment (Ksenaaa) [reply src/utils/parceStringToNumberAndPoint.ts]: fixed

File: src/component/radioButton/RadioButtonRegistration.tsx (hunk 16 above)
Comment (Ksenaaa) [reply src/component/radioButton/RadioButtonRegistration.tsx]: fixed

**Review by valentinPortside — APPROVED** [review]

**Pull request merged**